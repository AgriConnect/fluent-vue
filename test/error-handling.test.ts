import { createLocalVue, mount } from '@vue/test-utils'

import { FluentBundle, FluentResource } from '@fluent/bundle'
import ftl from '@fluent/dedent'

import FluentVue from '../src'

describe('vue integration', () => {
  const localVue = createLocalVue()
  localVue.use(FluentVue)

  const bundle = new FluentBundle('en-US', {
    useIsolating: false
  })

  bundle.addResource(
    new FluentResource(ftl`
    message = Hello, { $name }!
    sub-message = Hi, { $name }
  `)
  )

  const fluent = new FluentVue({
    bundle
  })

  const options = {
    fluent,
    localVue
  }

  it('warns about missing translation', () => {
    // Arrange
    const component = {
      template: "<div>{{ $t('message-not-found') }}</div>"
    }

    // Act
    const mounted = mount(component, options)

    // Assert
    expect(mounted).toMatchSnapshot()
  })
})
