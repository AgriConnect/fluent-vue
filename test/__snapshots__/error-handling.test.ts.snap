// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vue integration warns about missing translation 1`] = `<div>message-not-found</div>`;
